doc/feature/restAPI.md

```diff
@@ -0,0 +1,56 @@
+Publishing the REST API exposed by the HTTP Server
+============
+
+The http server exposes C++ objects via a REST API, in a JSON serialized format.
+The purpose of this document is to describe how this API request is implemented,
+and invoked.
+
+## Requirements
+
+* Define how to invoke the application REST API
+* Implement the JSON representation of the REST API
+
+## API
+
+Invoking a GET resquest on the /api url returns a JSON representation of the
+schemas for all registered serializables.
```

Comment:
I had something different in mind, and I it was more like what Django (the python implementation that we currently use for the Visualization Web Services) does. That is returning available URIs a as list when you do a GET on the /api uri of the server. What you are suggesting requires more processing on the client side, with a recursive walk in the tree of URIs. I would vote for a simple and single request for getting that list. But we can discuss again. This would simplify both the client and server side implementations.